src/app/embed/adapters/merge.adapter.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { ReplaySubject } from 'rxjs';
+import { QSDAdapter } from './qsd.adapter'
+import { FeedAdapter } from './feed.adapter'
+import { AdapterProperties, hasMinimumParams, DataAdapter, getMergedValues } from './adapter.properties'
+
+const NO_EMIT_YET = Symbol();
+
+@Injectable()
+export class MergeAdapter {
+  private adapters: DataAdapter[]
+  constructor(
+    private FeedAdapter: FeedAdapter,
+    private QSDAdapter:  QSDAdapter
+	) { 
+    this.adapters = [this.QSDAdapter, this.FeedAdapter]
+  }
+
+  getProperties(params): Observable<AdapterProperties> {
+    let chosenAdapters: Observable<AdapterProperties>[];
+
+    chosenAdapters = this.adapters.
+      map(obs => obs.getProperties(params)).
+      map(obs => obs.startWith(NO_EMIT_YET))
+
+    return Observable.combineLatest(...chosenAdapters).map(sources => {
+      let data = [];
+      for (let source of sources) {
+        if (source === NO_EMIT_YET) { break; }
+        data.push(source);
+      }
+      return getMergedValues(...data);
```

Comment:
then we return the merged result. We originally tried to use Object.assign but it overrides values with undefined so it didn't work well for our use case. So, to continue the story from above, if the values of 1, 2, 4, and 5 have come through, we emit the result of cascading the values of 1 on the values of 2. If 1-5 have emitted, we cascade all the way down, and if 1 hasn't emitted yet (unlikely since it's data from the Query String, but conceivable that it would be structured differently one day), it will always emit {}